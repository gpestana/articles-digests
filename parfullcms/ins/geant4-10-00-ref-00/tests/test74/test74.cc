//
// ********************************************************************
// * License and Disclaimer                                           *
// *                                                                  *
// * The  Geant4 software  is  copyright of the Copyright Holders  of *
// * the Geant4 Collaboration.  It is provided  under  the terms  and *
// * conditions of the Geant4 Software License,  included in the file *
// * LICENSE and available at  http://cern.ch/geant4/license .  These *
// * include a list of copyright holders.                             *
// *                                                                  *
// * Neither the authors of this software system, nor their employing *
// * institutes,nor the agencies providing financial support for this *
// * work  make  any representation or  warranty, express or implied, *
// * regarding  this  software system or assume any liability for its *
// * use.  Please see the license in the file  LICENSE  and URL above *
// * for the full disclaimer and the limitation of liability.         *
// *                                                                  *
// * This  code  implementation is the result of  the  scientific and *
// * technical work of the GEANT4 collaboration.                      *
// * By using,  copying,  modifying or  distributing the software (or *
// * any work based  on the software)  you  agree  to acknowledge its *
// * use  in  resulting  scientific  publications,  and indicate your *
// * acceptance of all terms of the Geant4 Software license.          *
// ********************************************************************
//
// -------------------------------------------------------------------
// $Id: test74.cc 74992 2013-10-25 10:44:55Z gcosmo $
// -------------------------------------------------------------------

//....oooOO0OOooo........oooOO0OOooo........oooOO0OOooo........oooOO0OOooo....

#include "G4RunManager.hh"
#include "G4MTRunManager.hh"
#include "G4UImanager.hh"
#include "G4UIterminal.hh"
#include "G4UItcsh.hh"

#ifdef G4VIS_USE
  #include "G4VisExecutive.hh"
#endif

#include "DetectorConstruction.hh"
#include "PhysicsList.hh"
#include "PrimaryGeneratorAction.hh"
#include "RunAction.hh"
#include "EventAction.hh"
#include "SteppingAction.hh"
#include "SteppingVerbose.hh"
#include "ActionInitialization.hh"

//....oooOO0OOooo........oooOO0OOooo........oooOO0OOooo........oooOO0OOooo....

int main(int argc,char** argv) 
{
  	// Choose the Random engine

	G4Random::setTheEngine(new CLHEP::RanecuEngine);

	// Construct the default run manager

#ifdef G4MULTITHREADED
	G4MTRunManager* runManager = new G4MTRunManager;
	runManager->SetNumberOfThreads(2);
#else
	G4RunManager* runManager = new G4RunManager;
#endif

  // Set mandatory user initialization classes
  DetectorConstruction* detector = new DetectorConstruction;
  runManager->SetUserInitialization(detector);
  
  PhysicsList * pl;
  runManager->SetUserInitialization(pl = new PhysicsList);
  pl->DisableCheckParticleList();
  
  // User action initialization
  //
  runManager->SetUserInitialization(new ActionInitialization());
 
  
  // Initialize G4 kernel
  //
  runManager->Initialize();
  
#ifdef G4VIS_USE
  G4VisManager* visManager = new G4VisExecutive;
  visManager->Initialize();
#endif
   
  // Get the pointer to the User Interface manager 
  G4UImanager* UI = G4UImanager::GetUIpointer();  

  if (argc==1)   // Define UI session for interactive mode.
  { 
#ifdef _WIN32
    G4UIsession * session = new G4UIterminal();
#else
    G4UIsession * session = new G4UIterminal(new G4UItcsh);
#endif
    UI->ApplyCommand("/control/execute test74.mac");    
    session->SessionStart();
    delete session;
  }
  else           // Batch mode
  { 
    G4String command = "/control/execute ";
    G4String fileName = argv[1];
    UI->ApplyCommand(command+fileName);
  }

#ifdef G4VIS_USE
  delete visManager;
#endif

  delete runManager;

  return 0;
}

