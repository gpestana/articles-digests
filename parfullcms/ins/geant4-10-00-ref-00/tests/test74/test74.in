#
# Macro file for test74
#
/control/verbose 2
/run/verbose 1
/run/initialize
/tracking/verbose 0
/process/list
/particle/list
/run/particle/dumpCutValues
#
/gun/direction 0 0 1
#
/gun/particle e-
/gun/energy 10000 eV
/run/beamOn 1
#
/gun/particle proton
/gun/energy 0.1 MeV
/run/beamOn 1
#
/gun/particle ion
/gun/ion 20 48 15
/gun/energy 4.8 MeV
/run/beamOn 1
#
exit
