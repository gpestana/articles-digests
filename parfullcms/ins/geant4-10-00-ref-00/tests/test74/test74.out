
        ############################################
        !!! WARNING - FPE detection is activated !!!
        ############################################

*************************************************************
 Geant4 version Name: geant4-10-00-ref-00    (6-December-2013)
                      Copyright : Geant4 Collaboration
                      Reference : NIM A 506 (2003), 250-303
                            WWW : http://cern.ch/geant4
*************************************************************

PhysicsList::SetCuts:CutLength : 1 nm 
Visualization Manager instantiating with verbosity "warnings (3)"...
Visualization Manager initialising...
Registering graphics systems...

You have successfully registered the following graphics systems.
Current available graphics systems are:
  ASCIITree (ATree)
  DAWNFILE (DAWNFILE)
  G4HepRep (HepRepXML)
  G4HepRepFile (HepRepFile)
  RayTracer (RayTracer)
  VRML1FILE (VRML1FILE)
  VRML2FILE (VRML2FILE)
  gMocrenFile (gMocrenFile)

Registering model factories...

You have successfully registered the following model factories.
Registered model factories:
  generic
  drawByCharge
  drawByParticleID
  drawByOriginVolume
  drawByAttribute

Registered filter factories:
  chargeFilter
  particleFilter
  originVolumeFilter
  attributeFilter

You have successfully registered the following user vis actions.
Run Duration User Vis Actions: none
End of Event User Vis Actions: none
End of Run User Vis Actions: none

Some /vis commands (optionally) take a string to specify colour.
Available colours:
  black, blue, brown, cyan, gray, green, grey, magenta, red, white, yellow

/run/verbose 1
/run/initialize
/tracking/verbose 0
/process/list
     Transportation,ion_G4MicroElecInelastic,e-_G4MicroElecElastic,e-_G4MicroElecInelastic
p_G4MicroElecInelastic
/particle/list
         GenericIon,                He3,              alpha,           anti_He3
         anti_alpha,      anti_deuteron,        anti_proton,        anti_triton
           deuteron,                 e+,                 e-,              gamma
           geantino,      opticalphoton,             proton,             triton

/run/particle/dumpCutValues
#
/gun/direction 0 0 1
#
/gun/particle e-
/gun/energy 10000 eV
/run/beamOn 1

e-_G4MicroElecElastic:   for  e-    SubType= 51
      ===== EM models for the G4Region  DefaultRegionForTheWorld ======
MicroElecElasticModel :  Emin=        0 eV    Emax=      100 MeV

e-_G4MicroElecInelastic:   for  e-    SubType= 53
      ===== EM models for the G4Region  DefaultRegionForTheWorld ======
MicroElecInelasticModel :  Emin=        0 eV    Emax=      100 MeV

p_G4MicroElecInelastic:   for  proton    SubType= 53
      ===== EM models for the G4Region  DefaultRegionForTheWorld ======
MicroElecInelasticModel :  Emin=        0 eV    Emax=       10 GeV

ion_G4MicroElecInelastic:   for  GenericIon    SubType= 53
      ===== EM models for the G4Region  DefaultRegionForTheWorld ======
MicroElecInelasticModel :  Emin=        0 eV    Emax=       10 TeV

========= Table of registered couples ==============================

Index : 0     used in the geometry : Yes
 Material : G4_Si
 Range cuts        :  gamma  1 nm     e-  1 nm     e+  1 nm  proton 1 nm 
 Energy thresholds :  gamma  990 eV     e-  990 eV     e+  990 eV  proton 0.1 eV 
 Region(s) which use this couple : 
    DefaultRegionForTheWorld

====================================================================

### Run 0 starts.
Run terminated.
Run Summary
  Number of events processed : 1
  User=0.21s Real=0.21s Sys=0s

 The run consists of 1 e- of 10 keV through 1 m   of G4_Si (density: 2.33 g/cm3 )

 Process calls frequency --->	e-_G4MicroElecElastic = 2072	e-_G4MicroElecInelastic = 395
 Total transfered energy (keV)=10

#
/gun/particle proton
/gun/energy 0.1 MeV
/run/beamOn 1

========= Table of registered couples ==============================

Index : 0     used in the geometry : Yes
 Material : G4_Si
 Range cuts        :  gamma  1 nm     e-  1 nm     e+  1 nm  proton 1 nm 
 Energy thresholds :  gamma  990 eV     e-  990 eV     e+  990 eV  proton 0.1 eV 
 Region(s) which use this couple : 
    DefaultRegionForTheWorld

====================================================================

### Run 1 starts.
Run terminated.
Run Summary
  Number of events processed : 1
  User=0.55s Real=0.55s Sys=0s

 The run consists of 1 proton of 100 keV through 1 m   of G4_Si (density: 2.33 g/cm3 )

 Process calls frequency --->	p_G4MicroElecInelastic = 1425	e-_G4MicroElecInelastic = 921	e-_G4MicroElecElastic = 11841
 Total transfered energy (keV)=100

#
/gun/particle ion
/gun/ion 20 48 15
/gun/energy 4.8 MeV
/run/beamOn 1

========= Table of registered couples ==============================

Index : 0     used in the geometry : Yes
 Material : G4_Si
 Range cuts        :  gamma  1 nm     e-  1 nm     e+  1 nm  proton 1 nm 
 Energy thresholds :  gamma  990 eV     e-  990 eV     e+  990 eV  proton 0.1 eV 
 Region(s) which use this couple : 
    DefaultRegionForTheWorld

====================================================================

### Run 2 starts.
Run terminated.
Run Summary
  Number of events processed : 1
  User=27.75s Real=27.61s Sys=0.05s

 The run consists of 1 Ca48 of 4.8 MeV through 1 m   of G4_Si (density: 2.33 g/cm3 )

 Process calls frequency --->	ion_G4MicroElecInelastic = 68243	e-_G4MicroElecElastic = 578566	e-_G4MicroElecInelastic = 43804
 Total transfered energy (keV)=4800

#
Graphics systems deleted.
Visualization Manager deleting...
G4 kernel has come to Quit state.
================== Deleting memory pools ===================
Number of memory pools allocated: 11 of which, static: 0
Dynamic pools deleted: 11 / Total memory freed: 33 Mb
============================================================
RunManagerKernel is deleted. Good bye :)
